### PR TITLE
DOC: Add clarifications to docs for setting up a development environment

### PR DESCRIPTION
**Pull request by smij720**
Hi all - having just been through the setup process (Mac, mamba) and as a non-expert I thought I'd feed back some things that would have made my experience faster and smoother.

---
**Change 1**
- It wasn't 100% clear to me that I did **_not_** need to use any information from the link `https://devguide.python.org/setup/#macos`  (it is clear with hindsight, but it's the initial impression that I'm trying to improve here):

Before this change:
<kbd>image</kbd>

After:
<kbd>image</kbd>

---
**Change 2**
- I got confused by `mamba` vs `conda`, and that you actually need to install is called `miniforge` - I had a moment of doubt here that I was on the wrong track.

Before this change:
<kbd>image</kbd>

<kbd>image</kbd>

---
**Change 3**
- I found that I had some old packages that were used in my mamba environment (probably from a previous install of mamba - last I played with this was about a year ago). Looking into them, they were packages like `sphinx` that did not have a version pinned in `environment.yml`. This feels like it could be a source of errors for people getting set up.

After:
<kbd>image</kbd>

---
**Change 4**
- Sounds strange, but I wasn't sure when I was actually 'done' with getting set up.
- I tend to like to run unit tests when I get set up with a new repo.
- I didn't realize that with this setup some of the unit tests (the ones with external dependencies) were expected to fail, as helpfully documented [here](https://pandas.pydata.org/docs/dev/development/contributing_codebase.html#running-the-test-suite).  I feel this is really important - I lost a lot of time because I assumed there must be something wrong with my install because unit tests were failing. By adding the link here hopefully others see this information sooner.

Before this change:
<kbd>image</kbd>

After:
<kbd>image</kbd>

---

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/development/contributing_environment.rst [hunk 1]
```diff
@@ -87,14 +88,24 @@ Option 1: using mamba (recommended)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 * Install `mamba <https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html>`_
+  (the instructions may direct you to install `miniforge <https://github.com/conda-forge/miniforge>`_,
+  which is OK - it includes mamba)
```
Comment (mroeschke) [inline doc/source/development/contributing_environment.rst:92]:
```suggestion
* Install miniforge to get `mamba <https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html>`_
```

File: doc/source/development/contributing_environment.rst [hunk 2]
```diff
@@ -87,14 +88,24 @@ Option 1: using mamba (recommended)
...
 * Make sure your mamba is up to date (``mamba update mamba``)
+* Create and activate the ``pandas-dev`` mamba environment using the following commands:
 
 .. code-block:: none
 
-   # Create and activate the build environment
    mamba env create --file environment.yml
    mamba activate pandas-dev
 
+* Ensure any python packages without a version specified in the ``environment.yml`` file
```
Comment (mroeschke) [inline doc/source/development/contributing_environment.rst:101]: This is not entirely necessary given the instructions describe a fresh install.

File: doc/source/development/contributing_environment.rst (hunk 1 above)
Comment (smij720) [reply doc/source/development/contributing_environment.rst]: Thank you - great suggestion

File: doc/source/development/contributing_environment.rst (hunk 2 above)
Comment (smij720) [reply doc/source/development/contributing_environment.rst:99]: Thanks.  Yes, agreed.  I'll remove this - it keeps it cleaner.

Comment (smij720) [reply doc/source/development/contributing_environment.rst]: Done.

File: doc/source/development/contributing_environment.rst (hunk 1 above)
Comment (smij720) [reply doc/source/development/contributing_environment.rst]: With this suggestion is now looks like this.  Confirmed link works.

<kbd>image</kbd>

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @smij720

**Pull request merged**